receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -79,5 +85,15 @@ func (s *scraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	}
 	s.mb.RecordSystemCPULogicalCountDataPoint(now, int64(numCPU))
 
+	cpuInfos, err := s.getCPUInfo()
+	if err != nil {
+		return pmetric.NewMetrics(), scrapererror.NewPartialScrapeError(err, metricsLen)
+	}
+	if len(cpuInfos) > 0 {
+		for _, cInfo := range cpuInfos {
+			s.mb.RecordSystemCPUFrequencyDataPoint(now, cInfo.frequency, fmt.Sprintf("cpu%d", cInfo.processor))
+		}
+	}
```

Comment:
If `len(cpuInfos)` is nil/0 its ignored anyway.
```suggestion
	for _, cInfo := range cpuInfos {
		s.mb.RecordSystemCPUFrequencyDataPoint(now, cInfo.frequency, fmt.Sprintf("cpu%d", cInfo.processor))
	}
```